lc-hax/Scripts/Patches/UnlimitedJumpPatch.cs

```diff
@@ -10,9 +10,9 @@
 class UnlimitedJumpPatch {
     static bool Prefix(PlayerControllerB __instance) {
         if (!Setting.EnableUnlimitedJump || !__instance.IsSelf()) return true;
-        if (__instance is { inSpecialInteractAnimation: true, isTypingChat: true, isPlayerControlled: false }) return true;
-        if (__instance.quickMenuManager.isMenuOpen) return true;
-
+        if (__instance is { inSpecialInteractAnimation: true, isTypingChat: true, isPlayerControlled: false }) return false;
+        if (__instance.quickMenuManager.isMenuOpen) return false;
+        if (__instance.isTypingChat) return false;
```

Comment:
@coderabbitai Can you write this better? My initial logic is wrong. I want this to return false if inSpecialInteractAnimation is true or isTypingChat is true or isPlayerController is false. This is using the latest C# syntax.

Reply:
Why not this?

```cs
if (__instance is { inSpecialInteractAnimation: true } or { isTypingChat: true } or { isPlayerControlled: false }) return false;
```